src/main/java/org/opensearch/neuralsearch/query/HybridQueryBuilder.java

```diff
@@ -263,7 +294,7 @@ protected boolean doEquals(HybridQueryBuilder obj) {
      */
     @Override
     protected int doHashCode() {
-        return Objects.hash(queries);
+        return Objects.hash(queries, paginationDepth);
```

Comment:
shouldn't you check for cluster version here? similar to equals

Reply:
is this driven primarily by code consistency since we have the same check in the equals method, or is there an actual scenario where this could be a problem? I can't think of anything other than rolling upgrades, and I'm not even sure if it would be an issue in that case.

Reply:
I think it would be good to put this check as it makes the code consistent in the entire file

Reply:
consistency come for `equals` method after you have added it to `hashcode` method. I'm asking why it was added to `hashcode` in the first place.